modules/transactions/src/main/java/org/apache/ignite/internal/tx/storage/state/TxStateStorage.java

```diff
@@ -122,9 +125,55 @@ public interface TxStateStorage extends ManuallyCloseable {
     void close();
 
     /**
-     * Removes all data from the storage.
+     * Closes and removes all data from the storage.
      *
-     * @throws StorageException In case when the operation has failed.
+     * @throws IgniteInternalException with {@link Transactions#TX_STATE_STORAGE_ERR} error code in case when the operation has failed.
      */
     void destroy();
+
+    /**
+     * Prepares the transaction state storage for a full rebalance: clears the storage, sets the {@link #lastAppliedIndex()} and
+     * {@link #lastAppliedTerm()} to {@link #FULL_REBALANCE_IN_PROGRESS}, and closes all cursors.
+     *
+     * <p>After calling this method, only write methods will be available, and read methods with {@link #lastApplied(long, long)} will
+     * throw {@link IgniteInternalException}.
```

Comment:
Can we have a more specific exception? Maybe a link to the corresponding error code? Let's make our exceptions better.